agr_java_core/src/main/java/org/alliancegenome/core/config/CacheConfig.java

```diff
@@ -0,0 +1,81 @@
+package org.alliancegenome.core.config;
+
+import org.alliancegenome.cache.CacheAlliance;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.configuration.cache.StorageType;
+import org.infinispan.eviction.EvictionType;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class CacheConfig {
+
+    private static RemoteCacheManager manager;
+    
+    public static RemoteCacheManager defaultRemoteCacheManager() {
+        if(manager != null) return manager;
```

Comment:
Does that need to be made thread-safe? I don't know how log it takes to create the manager object...

Reply:
The manager object is a singleton or you could say application scoped. However it's hooked up the the hotrod client which is thread safe from my reading.

Reply:
I still think this is not thread-safe but we'll see.